RavenDB-7069

talweiss1982 opened the pull request:
* added a logger to Async/AsyncServerClient.cs
* now fetching topology from the client in a sync manor to insure we have selected the correct request executor
* change client profiling to reflect on the underlining LRU implementation
* added force topology update for replication informer
* changed cluster topology cache and replication cache to have the same format
*  removed index replication test that is no longer valid, it was trying to prevent it from the client which is not supported, i made it deprecated.

Commit:
RavenDB-7069
* added a logger to Async/AsyncServerClient.cs
* now fetching topology from the client in a sync manor to insure we have selected the correct request executor
* change client profiling to reflect on the underlining LRU implementation
* added force topology update for replication informer
* changed cluster topology cache and replication cache to have the same format
*  removed index replication test that is no longer valid, it was trying to prevent it from the client which is not supported, i made it deprecated.

ayende commented:
@talweiss1982 tests are failing

ayende commented on Raven.Client.Lightweight/Connection/ReplicationInformerLocalCache.cs:
```diff
@@ -123,7 +123,7 @@ public static void ClearLocalCache(string serverHash)
         }
 
-        public static List<OperationMetadata> TryLoadClusterNodesFromLocalCache(string serverHash)
+/*        public static List<OperationMetadata> TryLoadClusterNodesFromLocalCache(string serverHash)
```
Why is this commented (and not deleted)?

talweiss1982 replied:
i forgot to delete it.

talweiss1982 commented:
I'll go over the failing tests don't modify the PR so we won't run into conflicts

Commit:
RavenDB-7069
Fixing tests that were broken due to the fact that we already fetched the topology thus the fail-over didn't change.
The fix was implemented in the previous PR just had to invoke the fetch topology with the force flag

Commit:
RavenDB-7069 forgot to delete commented code again :)
